HiringApp/HiringAppCore/Providers/CacheProvider.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  CacheProvider.swift
+//  Pods
+//
+//  Created by Alba Luján on 22/6/17.
+//
+//
+
+import Foundation
+import BSWFoundation
+import Deferred
+
+protocol CacheProviderType {
+    func getTechnologies() -> [TechnologyModel]?
+    func saveTechnologies(technologies: [TechnologyModel]) -> Task<()>
+}
+
+class CacheProvider: CacheProviderType {
+    
+    let defaults = UserDefaults.standard
+    let technologies: [TechnologyModel]? = nil
+    
+    func getTechnologies() -> [TechnologyModel]? {
+        guard let storedTechnologies = defaults.array(forKey: "technologies") as? [[String:Any]] else {
+            return nil
+        }
+        
+        var technologiesConverted = [TechnologyModel]()
+        storedTechnologies.forEach { (technology) in
+            let tech = TechnologyModel(id: technology["id"] as! Int, title: technology["title"] as! String, imageURL: technology["imageURL"] as! String, testAvailable: technology["testAvailable"] as! Bool, submittedTest: technology["submittedTest"] as? [String : String])
```

Comment:
I think it should be a good idea to parse the JSON to an object instead of passing by all the parameters extracted manually from a JSON